pkg/controller/intrusiondetection/intrusiondetection_controller.go

```diff
@@ -103,119 +112,92 @@ func Add(mgr manager.Manager, opts options.AddOptions) error {
-	if err != nil {
+	// Watch for changes to operator.tigera.io APIs.
+	if err = c.Watch(&source.Kind{Type: &operatorv1.IntrusionDetection{}}, eventHandler); err != nil {
 		return fmt.Errorf("intrusiondetection-controller failed to watch primary resource: %v", err)
 	}
-
-	err = c.Watch(&source.Kind{Type: &batchv1.Job{ObjectMeta: metav1.ObjectMeta{
-		Namespace: render.IntrusionDetectionNamespace,
-		Name:      render.IntrusionDetectionInstallerJobName,
-	}}}, &handler.EnqueueRequestForObject{})
-	if err != nil {
-		return fmt.Errorf("intrusiondetection-controller failed to watch installer job: %v", err)
-	}
-
-	// Watch for changes to to primary resource LogCollector, to determine if syslog forwarding is
-	// turned on or off.
-	err = c.Watch(&source.Kind{Type: &operatorv1.LogCollector{}}, &handler.EnqueueRequestForObject{})
-	if err != nil {
+	if err = c.Watch(&source.Kind{Type: &operatorv1.LogCollector{}}, eventHandler); err != nil {
 		return fmt.Errorf("intrusiondetection-controller failed to watch LogCollector resource: %v", err)
 	}
-
-	// Watch for changes to primary resource ManagementCluster
-	err = c.Watch(&source.Kind{Type: &operatorv1.ManagementCluster{}}, &handler.EnqueueRequestForObject{})
-	if err != nil {
+	if err = c.Watch(&source.Kind{Type: &operatorv1.ManagementCluster{}}, eventHandler); err != nil {
 		return fmt.Errorf("intrusiondetection-controller failed to watch primary resource: %w", err)
 	}
-
-	if err = utils.AddInstallationWatch(c); err != nil {
-		return fmt.Errorf("intrusiondetection-controller failed to watch Installation resource: %v", err)
+	if err = c.Watch(&source.Kind{Type: &operatorv1.Installation{}}, eventHandler); err != nil {
+		return fmt.Errorf("intrusiondetection-controller failed to watch Installation resource: %w", err)
 	}
-
-	if err = imageset.AddImageSetWatch(c); err != nil {
+	if err = c.Watch(&source.Kind{Type: &operatorv1.APIServer{}}, eventHandler); err != nil {
+		return fmt.Errorf("intrusiondetection-controller failed to watch APIServer resource: %w", err)
+	}
+	if err = c.Watch(&source.Kind{Type: &operatorv1.ImageSet{}}, eventHandler); err != nil {
 		return fmt.Errorf("intrusiondetection-controller failed to watch ImageSet: %w", err)
 	}
-
-	// Watch for changes in storage classes to queue changes if new storage classes may be made available for AD API.
-	if err = c.Watch(&source.Kind{Type: &storagev1.StorageClass{}}, &handler.EnqueueRequestForObject{}); err != nil {
-		return fmt.Errorf("intrusiondetection-controller failed to watch StorageClass resource: %w", err)
+	if err = utils.AddTigeraStatusWatch(c, tigeraStatusName); err != nil {
+		return fmt.Errorf("intrusiondetection-controller failed to watch intrusion-detection Tigerastatus: %w", err)
 	}
 
-	if err = utils.AddAPIServerWatch(c); err != nil {
-		return fmt.Errorf("intrusiondetection-controller failed to watch APIServer resource: %v", err)
+	err = c.Watch(&source.Kind{Type: &batchv1.Job{ObjectMeta: metav1.ObjectMeta{
```

Comment:
Yeah, I think that pattern is not appropriate for these. The only reason "AddSecretsWatch" is useful is because it does stuff with predicates, which we woudln't be doing here. I am mostly against adding helpers that don't save us any lines of code or repeated complexity.

In other places, where we already had these helper functions, it made the update to multi-tenancy a lot more tedious than it needed to be if they weren't there.